### PR TITLE
Add Dockerfile for building on Docker hub

### PR DESCRIPTION
**Pull request by robinst**
This is the first part for #65.

You can see the image here: https://hub.docker.com/r/robinst/tarpaulin/

You can try it out in a repository where you want to calculate coverage using this command:

    docker run --security-opt seccomp=unconfined -v "$PWD:/volume" robinst/tarpaulin

If you want more options, you have to run it like this:

    docker run --security-opt seccomp=unconfined -v "$PWD:/volume" robinst/tarpaulin cargo tarpaulin --ignore-tests --out Xml

If you want to run multiple commands, you have to use a shell like this:

    docker run --security-opt seccomp=unconfined -v "$PWD:/volume" robinst/tarpaulin sh -c "cargo build && cargo tarpaulin"

After merging this (or even before), you can set up an automated build in your account using "Create Automated Build" on Docker hub:

screen shot 2017-12-01 at 13 53 20

You will be able to configure when it should be built automatically, e.g. only build tags or also build master. You probably want to remove automatic building of all branches.

Once that's done I can raise another PR to add the instructions to the README :).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+RUN apt-get update && \
+    apt-get install -y cmake && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY . /opt/tarpaulin/
+
+RUN cd /opt/tarpaulin/ && \
+    cargo install && \
+    rm -rf /opt/tarpaulin/ && \
+    rm -rf /usr/local/cargo/registry/
```
Comment (robinst) [inline Dockerfile]: These are to make the image a bit smaller. And the registry would be out of date for users of this image anyway, between when we build it and they use it.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+WORKDIR /volume
```
Comment (robinst) [inline Dockerfile]: This is where the command is going to be run. The `-v "$PWD:/volume"` argument to `docker run` maps the working directory outside docker to `/volume`.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+CMD cargo build && cargo tarpaulin
```
Comment (robinst) [inline Dockerfile]: I think the `cargo build` is not necessary, but I've added it for now. See #67 for why.

**Comment by xd009642** [comment]
Why ever the beta and nightly builds fail is likely a separate issue entirely. Everything looks good so I'm going to merge this in and set up dockerhub, thank you :+1:

**Comment by robinst** [comment]
I saw that you tried building the tag here: https://hub.docker.com/r/xd009642/tarpaulin/builds/bgvliskm2842x9ygyglbamm/

That's not gonna work until the next tag which will contain the Dockerfile. For now, can you set it up to build master, until we want to release this?

**Comment by xd009642** [comment]
Yeah I've currently got a docker image built called latest which is based on the most recent commit on master https://hub.docker.com/r/xd009642/tarpaulin/builds/bggdjcshcsh6dgp2bhpgbhk/ if you want to start the PR for the updates to the readme/documentation that would be helpful :smile:

**Pull request merged**